lorien/UI/Palette.gd

```diff
@@ -0,0 +1,66 @@
+extends Control
+class_name Palette
+
+signal color_changed
+
+# -------------------------------------------------------------------------------------------------
+export var _default_button_number: int
+export var _palette_theme: Theme
+export var _color_picker_path: NodePath 
+export(String, "Lorien-Default","Ice-Cream") var _default_palette: String 
+
+onready var _button_grid: GridContainer = $ScrollContainer/GridContainer
+onready var _color_picker: ColorPicker = get_node(_color_picker_path)
+onready var _palette_button_theme_normal: StyleBoxFlat = _palette_theme.get_stylebox("normal", "Button")
+onready var _palette_button_theme_focus: StyleBoxFlat = _palette_theme.get_stylebox("focus", "Button")
+
+var _palette_button: PackedScene = preload("res://UI/Components/PaletteButton.tscn")
+var palettes: Dictionary = {"Lorien-Default": ["#FFFFFF","#808080","#000000","#31F776","#3E43F7","#BD332F","#DB6507","#CBF731","#9507DB"], 
+"Ice-Cream": ["#6B3E26","#FFC5D9","#C2F2D0","#FFFFFF","#FDF5C9","#FFCB85"]}
+var _focused_button: Button
+
+# -------------------------------------------------------------------------------------------------
+func _ready():
+	_instance_buttons(0, _default_button_number)
+	_color_picker.connect("color_changed", self, "_on_color_picker_change")
+	_set_palette_style(_default_palette)
+
+# -------------------------------------------------------------------------------------------------
+func _instance_buttons(min_range: int, max_range: int) -> void:
+	for btn in range(min_range, max_range):
+		_button_grid.add_child(_palette_button.instance())
+		var _current_button: Button = _button_grid.get_child(btn).get_child(0)
+		_current_button.connect("button_down", self, "_on_button_press", [_current_button])
+		_current_button["custom_styles/normal"] = _palette_button_theme_normal.duplicate()
+		_current_button["custom_styles/focus"] = _palette_button_theme_focus.duplicate()
+		_current_button["custom_styles/hover"] = _current_button["custom_styles/focus"]
+
+# -------------------------------------------------------------------------------------------------
+func _set_palette_style(palette: String):
+	for btn in range(0, min(_default_button_number, palettes[palette].size())):
```

Comment:
No need for a range function here: 

`for btn in min(_default_button_number, palettes[palette].size()): ...`